app/views/hq/sponsors/index.html.haml

```diff
@@ -0,0 +1,37 @@
+- if @sponsors.empty?
```

Comment:
Given we know who are users are and their knowledge / expectations of the application, I still don't feel this is appropriate. We are only likely to get the empty set at the applications initial creation.
ActiveRecord would most likely blow up with an exception rather than return an empty set in other circumstances. It is additional logic in the view that is just not needed. In turn this creates additional branches for testing.
If we really cannot let go that we need to guard against this, although I think we should, we should push the logic back up to the controller and set a flash.